elastic/kibana
# Favicon goodies

Fixes #65585
^ We'll need to spin off a separate issue to consider allowing custom (user uploaded) favicons.

## Summary

To better differentiate Kibana browser tabs from other Elastic proporties, this PR changes the favicon in the following ways:
1. For production mode, displays the monochrome version of the Elastic cluster [1]
2. For development mode [2a], displays a monochrome hammer icon in the style of the cluster [2b]
3. Changes the favicon fill color based upon the current operating system color theme via the `prefers-color-scheme` media query [3]
4. Removes unnecessary favicon files for unsupported browsers [4]

### Browser screenshots

#### Chrome
<img width="600" alt="Chrome" src="https://user-images.githubusercontent.com/446285/102916076-93075f00-4448-11eb-89f0-7b06a2facf20.png">

#### Firefox
<img width="600" alt="Firefox" src="https://user-images.githubusercontent.com/446285/102916126-aa464c80-4448-11eb-9ce1-fc251ea37c34.png">

#### Edge
<img width="600" alt="Edge" src="https://user-images.githubusercontent.com/446285/102916150-b500e180-4448-11eb-8a12-7dbd6e8ca3bb.png">

#### Safari
<img width="600" alt="Safari" src="https://user-images.githubusercontent.com/446285/102916164-ba5e2c00-4448-11eb-9774-81a813172870.png">

### How to test
1. Start up Kibana; see new favicon in your browser tab.
2. Change system theme, then refresh browser.
3. Change environment mode (I'm actually not sure how to do this, so I've only flipped the tertiary check. Is there a `yarn start` flag or something similar to alter this setting locally?)

### Checklist

Delete any items that are not applicable to this PR.


### Footnotes
[1] The monochrome version is included in the [updated branding guidelines](https://brand.elastic.co/302f66895/p/06c73c-elastic-logo/t/14994f)
[2a] As I understand it, a production instance can be run in dev mode and vice-versa. Regardless, I think it's beneficial to show the development favicon version based upon the mode as it is a clear, simple indicator of which mode Kibana is running under.
[2b] Since this is only for dev mode, it feels like an opportunity to have a little fun with the favicon. Feedback welcome; let me know of any potential concerns you might have in this regard.
[3] There a few ways to accomplish this effect, but the majority favored the [SVG](https://css-tricks.com/svg-favicons-and-all-the-fun-things-we-can-do-with-them/#dark-mode-support) [style](https://flaviocopes.com/dark-mode-favicon/) [approach](https://catalin.red/svg-favicon-light-dark-theme/) versus [using Javascript](https://css-tricks.com/how-to-create-a-favicon-that-changes-automatically/). After reading through those and testing some things out, I chose the SVG with embedded `style` approach as it avoids the need for Javascript and reduces the number of graphic files needed. Also, it just plain worked with our existing setup.
[4] The major browsers, supported by Kibana, [support SVG favicons with the exception of Safari](https://caniuse.com/link-icon-svg). A fallback, black PNG favicon has been included to accommodate Safari (see screenshot above) and, as a side effect, likely works for several other browsers. For more information on favicon files, [this article was very helpful](https://css-tricks.com/favicon-quiz/).
<img width="310" alt="Screen Shot 2020-12-22 at 9 09 14 AM" src="https://user-images.githubusercontent.com/446285/102913853-fd1e0500-4444-11eb-9cbe-92f71094c197.png">
 

cc:/ @alexfrancoeur @elastic/kibana-design
